fars/api/views.py

```diff
@@ -11,18 +12,28 @@ class BookingFilter(filters.FilterSet):
     bookable = filters.CharFilter(field_name='bookable__id_str')
     before = filters.IsoDateTimeFilter(field_name='start', lookup_expr='lte')
     after = filters.IsoDateTimeFilter(field_name='end', lookup_expr='gte')
+    booking_group = filters.CharFilter(field_name='booking_group')
 
     class Meta:
         model = Booking
-        fields = ['bookable', 'before', 'after']
+        fields = ['bookable', 'before', 'after', 'booking_group']
 
 
 class BookingsList(viewsets.ViewSetMixin, generics.ListAPIView):
     permission_classes = [IsAuthenticated]
     queryset = Booking.objects.all()
     serializer_class = NoMetaBookingSerializer # Exclude metadata to hide doorcode in this API
-    filter_backends = (filters.DjangoFilterBackend,)
+    filter_backends = (filters.DjangoFilterBackend, SearchFilter, OrderingFilter)
     filter_class = BookingFilter
+    search_fields = ['comment', 'start', 'end', 'user__username', 'user__first_name', 'user__last_name', 'booking_group__name']
+    ordering_fields = ['start', 'end', 'id']
+    ordering = ['start', 'end']
+
+    def finalize_response(self, request, response, *args, **kwargs):
+        limit = request.query_params.get("limit")
+        if limit and limit.isdigit():
+            response.data = response.data[:int(limit)]
+        return super().finalize_response(request, response, *args, **kwargs)
```

Comment:
Ah, clicked approve before I had properly read it.
I think you ought to apply the limit in the DB query. Otherwise if the result set is large, the API call will be slow even if only a few results are returned in the response.

Reply:
I can not find a way to do the slicing after filtering and ordering. Any suggestions?

Reply:
https://www.django-rest-framework.org/api-guide/pagination/